cfscrape/scraper.py

```diff
@@ -1,20 +1,20 @@
 # -*- coding: utf-8 -*-
 
 import logging
-import re
 import ssl
 import subprocess
 import copy
 import time
 
-from base64 import b64encode
 from collections import OrderedDict
 
 from requests.sessions import Session
 from requests.compat import urlparse, urlunparse
 
 from .adapters import CloudflareAdapter
 from .exceptions import CloudflareCaptchaError
+from .challenges import CloudflareChallenge
```

Comment:
@Anorov 

This should make it easy to add another challenge type i.e. `CaptchaChallenge` which could be the base of `CloudflareCaptchaChallenge` and possibly a [hCaptcha](https://hcaptcha.com) class. Or perhaps just a `BaseChallenge`...

Parsing the siteKey and form values from the captcha page so a user can solve and submit would be an awesome feature.

Reply:
What scenarios did you have in mind for supporting alternative captchas like hCaptcha? As far as I know, Cloudflare only uses reCAPTCHA, and I don't plan on adding support for services other than Cloudflare. Maybe I'm misunderstanding something.

Reply:
No, you're right. I really meant to just note that adding other challenge types will be easier.

Aside from that, I was thinking that there wouldn't be much difference between the two but I really don't know. I know that https://github.com/privacypass/challenge-bypass-extension supports both and the extension itself seems to be supported/encouraged by Cloudflare. I don't know that I've ever encountered a hCaptcha and didn't have it in mind to personally add support for it.

Reply:
Example class: https://gist.github.com/pro-src/de7d48eb20839cb35febf55a6758e965